src/ng/compile.js

```diff
@@ -347,6 +346,36 @@
  *   and values into the expression wrapper fn. For example, if the expression is `increment(amount)`
  *   then we can specify the amount value by calling the `localFn` as `localFn({amount: 22})`.
  *
+ * All 4 kinds of bindings (`@`, `=`, `<`, and `&`) can be made optional by adding `?` to the expression.
+ * The marker must come before the attribute name.
+ * See the {@link error/$compile/iscp the Invalid Isolate Scope Definition error} for definition examples.
+ * This is useful to refine the interface directives provides.
+ * One subtle difference between optional and non-optional happens **when the binding attribute is not
+ * set**:
+ * - the binding is optional: the property will not be defined
+ * - the binding is not optional: the property is defined
+ *
+ * ```js
+ *app.directive('testDir', function() {
+    return {
+      scope: {
+        notoptional: '=',
+        optional: '=?',
+      },
+      bindToController: true,
```

Comment:
Nope: https://github.com/angular/angular.js/pull/15110
This is actually quite powerful ...